csrc/codegen.cpp

```diff
@@ -272,8 +273,17 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
   }
 
   // Generates the kernel function declaration
-  void genDeclaration(const std::string& kernel_name) {
+  void genDeclaration(
+      const std::string& kernel_name,
+      std::optional<int64_t> num_threads_per_cta) {
     code_ << "__global__ void ";
+    if (kernel_->hasManaged("warp_specialized_num_registers")) {
```

Comment:
See https://github.com/NVIDIA/Fuser/pull/3669/files#r1903613209, let me know if it makes sense to always unconditionally enable/disable `__launch_bounds__` depending on if the block size is static. This means, we should not check managed value here.

Reply:
I'm not sure of the performance impact of always applying `__launch_bounds__`. 

I believe it is similar to setting `maxrregcount`. e.g., `maxrregcount` means you set the maximum number of registers to get a certain cta size. `maxThreadsPerBlock` restricts the cta size so the maximum number of registers is bounded.